### PR TITLE
Plotter Refactor to not use `search_internal`

### PR DESCRIPTION
**Pull request by Jammy2211**
Visualization previous required to use of a `search_internal` object, where the internal representation of a search (e.g. `dynesty`, `emcee`) was used to produce visualizaiton.

This had a number of nasty consequences:

- The `search_internal` was passed around the `Samples` object alongside the samples themselves, leading to a lot of duplication.
- The internal representation of the search had to be stored to hard-disk, making large file sizes and the database interface less clean.
- Dependency issues with visualization libaries.

This PR makes it so plotters now only make plots which use the native PyAutoFit `Samples` objects, therefore removing all need to wrap `search_internal`. 

This simplifies the code and means hard disk space is saved by disabling the `search_internal` output storage after a fit is complete.

A number of plot types were removed in this PR, they will soon be readded in a way that interfaces directly with PyAutoFit samples

### CODE REVIEW

File: autofit/non_linear/plot/optimize_plotters.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import logging
+import numpy as np
+
+from autofit.non_linear.plot.samples_plotters import SamplesPlotter
+
+logger = logging.getLogger(__name__)
+
+class OptimizePlotter(SamplesPlotter):
+
+    pass
```
Comment (rhayes777) [inline autofit/non_linear/plot/optimize_plotters.py:11]: Also seemingly pointless?

File: autofit/non_linear/plot/samples_plotters.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    -------
+        A function that plots a visual, or None if test mode is on.
+    """
+
+    @wraps(func)
+    def wrapper(
+        obj: object,
+        *args,
+        **kwargs
+    ):
```
Comment (rhayes777) [inline autofit/non_linear/plot/samples_plotters.py:30]: If you're not using the obj argument explicitly then you can make a more versatile decorator with just *args and **kwargs

File: autofit/non_linear/plot/samples_plotters.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+        corner.corner(
+            data=np.asarray(self.samples.parameter_lists),
+            weight_list=self.samples.weight_list,
+            labels=self.model.parameter_labels_with_superscripts_latex,
+            **kwargs
+        )
+
+        self.output.to_figure(structure=None, auto_filename="corner")
+        self.close()
```
Comment (rhayes777) [inline autofit/non_linear/plot/samples_plotters.py:117]: Could use your decorator?

File: autofit/non_linear/search/abstract_search.py [hunk 4]
```diff
@@ -721,8 +722,19 @@ def result_via_completed_fit(
...
+        try:
+            search_internal = self.backend
+        except AttributeError:
+            search_internal = None
+
+        if search_internal is None:
+            try:
+                search_internal = self.paths.load_search_internal()
+            except FileNotFoundError:
+                search_internal = None
```
Comment (rhayes777) [inline autofit/non_linear/search/abstract_search.py:734]: Put this in the except

Comment (rhayes777) [reply autofit/non_linear/search/abstract_search.py:734]: Oh unless you expect self.backend to be None?

Comment (rhayes777) [reply autofit/non_linear/search/abstract_search.py:734]: Actually it's kind of odd to catch an attribute exception for an attribute of self

File: autofit/non_linear/search/nest/abstract_nest.py [hunk 5]
```diff
@@ -64,3 +66,19 @@ def config_type(self):
...
+        return NestPlotter
+
+    def plot_results(self, samples):
+
+        def should_plot(name):
+            return conf.instance["visualize"]["plots_search"][self.__class__.__name__][name]
+
+        plotter = self.plotter_cls(
+            samples=samples,
+            output=Output(path=self.paths.image_path / "search", format="png"),
```
Comment (rhayes777) [inline autofit/non_linear/search/nest/abstract_nest.py:81]: Oh my are you using pathlib!? :O

**Review by rhayes777 — APPROVED** [review]

File: autofit/config/visualize/plots_search.yaml [hunk 6]
```diff
@@ -1,26 +1,14 @@
...
+Emcee:
+  corner: true             # Output corner figure (using corner.py) during a non-linear search fit?
+Nautilus:
+  corner: true             # Output corner figure (using anestetic) during a non-linear search fit?
+PySwarmsLobal:
+PySwarmsGlobal:
+UltraNest:
+  corner: true             # Output corner figure (using anestetic) during a non-linear search fit?
+Zeus:
+  corner: true             # Output corner figure (using corner.py) during a non-linear search fit?
```
[the remark this answers is not in the document]
Comment (Jammy2211) [reply autofit/config/visualize/plots_search.yaml:14]: Yep, I have thought about doing that, and likely will.

File: autofit/non_linear/plot/mcmc_plotters.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import matplotlib.pyplot as plt
+import logging
+
+from autofit.non_linear.plot.samples_plotters import SamplesPlotter
+
+logger = logging.getLogger(__name__)
+
+class MCMCPlotter(SamplesPlotter):
+
+    pass
```
[the remark this answers is not in the document]
Comment (Jammy2211) [reply autofit/non_linear/plot/mcmc_plotters.py:10]: Once I add more plots it'll be extended, it did have functions but I removed them for simplicity for now.

**Pull request merged**